control-plane/agents/src/bin/core/volume/snapshot_operations.rs

```diff
@@ -432,13 +457,22 @@ impl OperationGuardArc<VolumeSnapshot> {
 async fn destroy_transactions(
     data_set: &HashMap<SnapshotTxId, Vec<ReplicaSnapshot>>,
     registry: &Registry,
+    max_prune_limit: Option<u32>,
 ) -> HashMap<SnapshotTxId, Vec<ReplicaSnapshot>> {
     // Create a map to track of transactions upon cleanup failure.
     let mut final_transactions: HashMap<SnapshotTxId, Vec<ReplicaSnapshot>> = HashMap::new();
+    // Track the cleanup.
+    let mut cleanup_count = 1;
 
     // Iterate over current transactions and destroy each replica snapshot in the given data set
     // transactions and store the transactions and replica snapshots whose cleanup failed.
     for (txn_id, snap_list) in data_set {
+        if let Some(limit) = max_prune_limit {
+            if cleanup_count >= limit {
+                break;
+            }
+            cleanup_count += 1;
+        }
```

Comment:
instead you can use .take on the data_set ?

Reply:
Seems like there is no such method for data_set, i.e a HashMap?